@AH-64D Apache Longbow/Addons/fza_ah64_fire/functions/fn_update.sqf

```diff
@@ -0,0 +1,81 @@
+Private _engineApuFire      = _heli getVariable "fza_ah64_apu_fire";
+
+if !_battBusOn exitwith {
+    _heli setVariable ["fza_ah64_fireArmed1", [false, 0, CBA_missionTime], true];
+    _heli setVariable ["fza_ah64_fireArmed2", [false, 0, CBA_missionTime], true];
+    _heli setVariable ["fza_ah64_fireArmedApu", [false, 0, CBA_missionTime], true];
+    //FIRE Arm
+    _heli setObjectTexture ["in_lt_fire1rdy", ""];
+    _heli setObjectTexture ["in_lt_fire2rdy", ""];
+    _heli setObjectTexture ["in_lt_fireapurdy", ""];
+    //FIRE Detected
+    _heli setObjectTexture ["in_lt_fire1", ""];
+    _heli setObjectTexture ["in_lt_fire2", ""];
+    _heli setObjectTexture ["in_lt_fireapu", ""];
+    //Fire bottle Dispensed
+    _heli setObjectTexture ["in_lt_firepdis", ""];
+    _heli setObjectTexture ["in_lt_firerdis", ""];
+};
+
+if (_engineOneArm #0 && _engineOneArm #2 <= CBA_missionTime - 2) then {
+    [_heli, "fza_sfmplus_engState", 0, "OFF", true] call fza_fnc_setArrayVariable;
+};
+if (_engineTwoArm #0 && _engineTwoArm #2 <= CBA_missionTime - 2) then {
+    [_heli, "fza_sfmplus_engState", 1, "OFF", true] call fza_fnc_setArrayVariable;
+};
+if (_engineApuArm #0 && _engineApuArm #2 <= CBA_missionTime - 2) then {
+    _heli setVariable ["fza_systems_apuBtnOn", false];
+};
+
+//Fire test switch
+if (fza_ah64_firetest == 1) exitwith {
+    _heli setobjecttexture ["in_lt_fire1", "\fza_ah64_us\tex\in\pushbut.paa"];
+    _heli setobjecttexture ["in_lt_fire2", "\fza_ah64_us\tex\in\pushbut.paa"];
+    _heli setobjecttexture ["in_lt_fireapu", "\fza_ah64_us\tex\in\pushbut.paa"];
+};
+if (fza_ah64_firetest == 2) exitwith {
+    _heli setobjecttexture ["in_lt_fire1", "\fza_ah64_us\tex\in\pushbut.paa"];
+    _heli setobjecttexture ["in_lt_fire2", "\fza_ah64_us\tex\in\pushbut.paa"];
+    _heli setobjecttexture ["in_lt_fireapu", "\fza_ah64_us\tex\in\pushbut.paa"];
+    _heli setobjecttexture ["in_lt_firepdis", "\fza_ah64_us\tex\in\pushbut.paa"];
+    _heli setobjecttexture ["in_lt_firerdis", "\fza_ah64_us\tex\in\pushbut.paa"];
+};
+
+//FIRE Arm
+_heli setObjectTexture ["in_lt_fire1rdy", ["", "\fza_ah64_us\tex\in\pushbut.paa"] select _engineOneArm #0];
+_heli setObjectTexture ["in_lt_fire2rdy", ["", "\fza_ah64_us\tex\in\pushbut.paa"] select _engineTwoArm #0];
+_heli setObjectTexture ["in_lt_fireapurdy", ["", "\fza_ah64_us\tex\in\pushbut.paa"] select _engineApuArm #0];
+//FIRE Detected
+_heli setObjectTexture ["in_lt_fire1", ["", "\fza_ah64_us\tex\in\pushbut.paa"] select _engineOneFire];
+_heli setObjectTexture ["in_lt_fire2", ["", "\fza_ah64_us\tex\in\pushbut.paa"] select _engineTwoFire];
+_heli setObjectTexture ["in_lt_fireapu", ["", "\fza_ah64_us\tex\in\pushbut.paa"] select _engineOneFire];
+//Fire bottle Dispensed
+_heli setObjectTexture ["in_lt_firepdis", ["", "\fza_ah64_us\tex\in\pushbut.paa"] select _PrimaryFBDispensed];
+_heli setObjectTexture ["in_lt_firerdis", ["", "\fza_ah64_us\tex\in\pushbut.paa"] select _ReserveFBDispensed];
```

Comment:
Rather than constantly `setObjectTexture`ing throughout this function, please can you instead have a variable that you set to true or false for each light and then use the bottom area here to see the textures once?